types/redux-actions/index.d.ts

```diff
@@ -148,13 +153,13 @@ export function handleAction<State, Payload>(
     actionType: string | ActionFunctions<Payload> | CombinedActionType,
     reducer: Reducer<State, Payload> | ReducerNextThrow<State, Payload>,
     initialState: State
-): Reducer<State, Payload>;
+): ReduxReducer<State, Payload>;
```

Comment:
@oddui what I don't get - why reducer type have to have state as undefined yet initialState is always defined?

Reply:
Hi @alexey-pelykh the return type here allows `undefined` as state, in which case it will pass initialState to the internal `reducer`. The internal `reducer`'s type is not changed `(state: State, action: Action<Payload>) => State`. Does this answer your question?

Reply:
Yes, thanks